lib/utils/aws/aws.go

```diff
@@ -484,3 +490,30 @@ func iamResourceARN(partition, accountID, resourceType, resourceName string) str
 		Resource:  fmt.Sprintf("%s/%s", resourceType, resourceName),
 	}.String()
 }
+
+// MaybeHashRoleSessionName truncates the role session name and adds a hash
+// when the original role session name is greater than AWS character limit
+// (64).
+func MaybeHashRoleSessionName(roleSessionName string) (ret string) {
+	if len(roleSessionName) <= MaxRoleSessionName {
+		return roleSessionName
+	}
+
+	hash := sha1.New()
+	hash.Write([]byte(roleSessionName))
+	hex := hex.EncodeToString(hash.Sum(nil))
+
+	// "1" for the delimiter.
+	keepPrefixIndex := MaxRoleSessionName - len(hex) - 1
+
+	if keepPrefixIndex < 0 {
+		// This should not happen since sha1 length and MaxRoleSessionName are
+		// both constant.
+		ret = hex[:MaxRoleSessionName]
```

Comment:
Hypothetically, if we hit this, `MaxRoleSessionName` could still be larger than `len(hex)`, which would make this line panic.

I would remove this branch and keep the suggestion above.